### PR TITLE
Document Dictionary.{Remove,Clear} enumeration safety

### PR DESCRIPTION
**Pull request by roji**
See #42123, #42212

Some trailing whitespace has been trimmed, let me know if that's an issue.

/cc @danmosemsft @stephentoub @ajcvickers

### CODE REVIEW

**Comment by jkotas** [comment]
This behavior is new with .Net Core 3. Could you please note this in the docs?

**Comment by roji** [comment]
Sure. Do we have some way of showing different content for different versions (based on the version chosen in the dropdown)? Or do we need to integrate version-specific information as notes?

/cc @mairaw @carlossanlop

**Comment by carlossanlop** [comment]
@roji I haven't seen an option in dotnet-api-docs that allows showing a description based on the version. What I have seen is something like:

> .NET Framework only: Behavior description.
> .NET Core 3.0 only: Behavior description.

**Comment by roji** [comment]
Updated to mention .NET Core 3.0 only.

File: xml/System.Collections.Generic/Dictionary`2+Enumerator.xml [hunk 1]
```diff
@@ -65,7 +65,9 @@
...
   
- An enumerator remains valid as long as the collection remains unchanged. If changes are made to the collection, such as adding, modifying, or deleting elements, the enumerator is irrecoverably invalidated and the next call to <xref:System.Collections.Generic.Dictionary%602.Enumerator.MoveNext%2A> or <xref:System.Collections.Generic.Dictionary%602.Enumerator.System%23Collections%23IEnumerator%23Reset%2A> throws an <xref:System.InvalidOperationException>.  
+ An enumerator remains valid as long as the collection remains unchanged. If changes are made to the collection, such as adding elements or changing the capacity, the enumerator is irrecoverably invalidated and the next call to <xref:System.Collections.Generic.Dictionary%602.Enumerator.MoveNext%2A> or <xref:System.Collections.Generic.Dictionary%602.Enumerator.System%23Collections%23IEnumerator%23Reset%2A> throws an <xref:System.InvalidOperationException>.
+
+.NET Core 3.0 only: The only mutating methods which do not invalidate enumerators are <xref:System.Collections.Generic.Dictionary%602.Remove%2A> and <xref:System.Collections.Generic.Dictionary%602.Clear%2A>.
```
Comment (jkotas) [inline xml/System.Collections.Generic/Dictionary`2+Enumerator.xml]: Nit: `.NET Core 3.0+` to make it clear that this applies to future .NET Core versions as well?

Comment (roji) [reply xml/System.Collections.Generic/Dictionary`2+Enumerator.xml]: Makes sense, done.

**Pull request merged**